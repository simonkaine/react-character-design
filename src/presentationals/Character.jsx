export default function Character({head, torso, legs}) {
    return (
        <section style={{backgroundColor: 'black', display: 'flex', flexDirection: 'column'}}>
            <article aria-label='head' style={{
                backgroundImage: `url(./${head}-head.png)`,
                margin: '0 auto',
                height: 300,
                width: 300,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>

            <article aria-label='torso' style={{
                backgroundImage: `url(./${torso}-torso.png)`,
                margin: '0 auto',
                height: 300,
                width: 300,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>
           
           <article aria-label='legs'style={{
                backgroundImage: `url(./${legs}-legs.png)`,
                margin: '0 auto',
                height: 300,
                width: 300,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>

        </section>
    )
}