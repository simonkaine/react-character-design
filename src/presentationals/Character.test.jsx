import {screen, render} from '@testing-library/react';
import Character from './Character.jsx';

it('Should display the Character from initial load', () => {
  const {container} = render(<Character head='dragon' torso='cena' legs='bird'/>);

  const head = screen.getByLabelText(/head/i);
  const torso = screen.getByLabelText(/torso/i);
  const legs = screen.getByLabelText(/legs/i);

  expect(head).toHaveStyle(`background-image: url(./dragon-head.png)`);
  expect(torso).toHaveStyle(`background-image: url(./cena-torso.png)`);
  expect(legs).toHaveStyle(`background-image: url(./bird-legs.png)`);
  expect(container).toMatchSnapshot();
});