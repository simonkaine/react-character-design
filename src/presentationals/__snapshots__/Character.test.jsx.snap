// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display the Character from initial load 1`] = `
<div>
  <section
    style="background-color: black; display: flex; flex-direction: column;"
  >
    <article
      aria-label="head"
      style="background-image: url(./dragon-head.png); margin: 0px auto; height: 300px; width: 300px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="torso"
      style="background-image: url(./cena-torso.png); margin: 0px auto; height: 300px; width: 300px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="legs"
      style="background-image: url(./bird-legs.png); margin: 0px auto; height: 300px; width: 300px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
